### PR TITLE
feat: allow customizing url, headers, http methods

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**

### CODE REVIEW

File: async_processor/types.py [hunk 1]
```diff
@@ -58,13 +73,25 @@ def get_body(self) -> Any:
...
+        return self.request_method
+
 
 # We cannot maintain two different types and should remove `InputMessage`
 # after sometime
 class InputMessageV2(InputMessageInterface):
     tfy_request_id: Optional[constr(regex=r"^[a-zA-Z0-9\-]{1,36}$")] = None
     tfy_published_at_epoch_ns: Optional[int] = None
     tfy_stream_response: bool = False
+    tfy_request_headers: Optional[Dict[str, Union[str, List[str]]]] = None
```
Comment (chiragjn) [inline async_processor/types.py:92]: Header values can be list? 🤔

Comment (debajyoti-truefoundry) [reply async_processor/types.py:92]: One header can have multiple values. I will probably change the type later.

File: async_processor/sidecar/sidecar.py [hunk 2]
```diff
@@ -131,7 +133,8 @@ async def _stream(
         self, input_message: InputMessageInterface
     ) -> AsyncIterator[OutputMessage]:
         async with self._client_session.post(
```
Comment (chiragjn) [inline async_processor/sidecar/sidecar.py:135]: `async with self._client_session.request(...)`

Comment (chiragjn) [inline async_processor/sidecar/sidecar.py:135]:
```suggestion
        async with self._client_session.request(
```

Comment (chiragjn) [reply async_processor/sidecar/sidecar.py]: can send headers here too?

**Review by chiragjn — APPROVED** [review]

File: async_processor/sidecar/sidecar.py [hunk 3]
```diff
@@ -86,6 +87,21 @@ async def iter(self) -> AsyncIterator[StreamBatch]:
...
+    if not url_from_message:
+        return settings.destination_url
+
+    parsed_base_url = urlparse(settings.destination_url)
+    parsed_url_from_message = urlparse(url_from_message)
+
+    return parsed_url_from_message._replace(
+        netloc=parsed_base_url.netloc,
+        scheme=parsed_base_url.scheme,
+    ).geturl()
```
Comment (chiragjn) [inline async_processor/sidecar/sidecar.py:102]: Is this like a security feature to not allow sending  request to arbitrary hosts?

**Review by chiragjn — APPROVED** [review]

**Review by chiragjn — APPROVED** [review]

File: async_processor/sidecar/sidecar.py (hunk 3 above)
Comment (debajyoti-truefoundry) [reply async_processor/sidecar/sidecar.py:102]: Let me discuss this offline with you.

**Pull request merged**